### ISSUE TITLE
Feature Request: New command(s) to remove styling

### ISSUE THREAD

**Issue by hmelman** [jrblevin/markdown-mode#283]
I wanted to remove bolding from some text. It would be nice if there was a command to do so. I'm not sure how it would work. 

1. Should it turn bold text into italics or straight to plain text? Maybe a user option to choose.
2. I'd be fine having to highlight all the (possibly multi-word) text to remove formatting though currently having point on a word without a region and using `C-c C-s b` will bold the whole word and leave point where it is. It would be nice for a remove function to have symmetry with this behavior.
3. I originally thought of a single command (say on `C-c C-r`) that would remove paired formatting (bold, italics, headings, code, etc). Perhaps it would be better to have the existing commands be a toggle. Currently if point is in a bolded word, `C-c C-s b` will add more asterisks which probably isn't what is wanted. Perhaps a toggle is more efficient behavior and works better in nested circumstances (this text is bold and code, remove the code tags) . This could also be easier to implement because code could know what specific chars to look for rather than working on the inner most nested styling.

**Comment by jrblevin** [comment]
Most of the markup insertion/styling commands work as toggles already, including bold.

> currently having point on a word without a region and using C-c C-s b will bold the whole word and leave point where it is. It would be nice for a remove function to have symmetry with this behavior.

That's the intended behavior of the toggle functionality. If it's not working like that, can you give a test case or instructions to replicate?

> Currently if point is in a bolded word, C-c C-s b will add more asterisks which probably isn't what is wanted.

If so, then I think that would be a bug.

**Comment by hmelman** [comment]
Took a little to narrow it down. Create a buffer with two list items:
```markdown
     - [foo](bar) - 
     - [**abc**](def) -
```
The important point is that 'abc' must be bolded. Put point in foo after the 'f'. `C-c C-s b` repeatedly will add asterisks. If point is between the two o's in foo it will behave more oddly.

**Comment by jrblevin** [comment]
Thanks for narrowing this down. I can reproduce what you described and will look into it.

**Comment by sten0** [comment]
Confirmed bug in toggle functionality, following @hmelman's steps to reproduce,
> If point is between the two o's in foo it will behave more oddly.

Transforms
```
- [foo](bar) - 
- [foo](bar) - 
- [**abc**](def) -
```

to (first interation with point in between the two `o`s)
```
- [foo](bar) - 
- [**foo**](bar) - 
- [**abc**](def) -
```

to (second interation with point in between the two `o`s)
```
- [foo](bar) - 
- [**foo](bar) - 
- [abc**](def) -
```

with Emacs 26.1 and markdown-mode f3c54e34cc5228001af36a5301883325319f21d4

**Comment by syohex** [comment]
I have merged #466. Please check latest version.

**Comment by sten0** [comment]
Thank you @syohex !  Confirmed fixed in c15f67f with Emacs 26.1
